### PR TITLE
feat: support column ranges in string format in use_columns 

### PR DESCRIPTION
**Pull request by lukapeschke**
As discussed, follow-up of #189

### CODE REVIEW

File: python/fastexcel/__init__.py [hunk 1]
```diff
@@ -171,8 +173,10 @@ def load_sheet_by_idx(
         :param schema_sample_rows: Specifies how many rows should be used to determine
                                    the dtype of a column.
                                    If `None`, all rows will be used.
-        :param use_columns: Specifies the columns to use. Can either be a list of column names, or
-                            a list of column indices (starting at 0).
+        :param use_columns: Specifies the columns to use. Can either be a list of column names, a
+                            a list of column indices (starting at 0), or a list of columns and
+                            ranges as column names indices in excel form. For example, `"A,C:E,F"`
```
Comment (PrettyWood) [inline python/fastexcel/__init__.py:178]: no it should be inclusive both ways as an excel user

File: src/types/excelsheet.rs [hunk 2]
```diff
@@ -149,28 +149,162 @@ impl SelectedColumns {
...
+
+        if col.is_empty() {
+            return Err(InvalidParameters(
+                "a column should have at least one character, got none".to_string(),
+            )
+            .into());
+        }
+
+        col.chars()
+            //  iterating over all chars reversed, to have a power based on their rank
```
Comment (PrettyWood) [inline src/types/excelsheet.rs:190]: smart

File: src/types/excelsheet.rs [hunk 3]
```diff
@@ -149,28 +149,162 @@ impl SelectedColumns {
...
+            Err(InvalidParameters(format!(
+                "expected range to contain exactly 2 elements, got {n_elements}: \"{col_range}\"",
+                n_elements = col_elements.len()
+            ))
+            .into())
+        }
+    }
+}
+
+// Does not support columns beyond 26 for now
```
Comment (PrettyWood) [inline src/types/excelsheet.rs:252]: It seems like you did the code to support it 🤔

File: src/types/excelsheet.rs [hunk 4]
```diff
@@ -590,10 +725,63 @@ mod tests {
...
+    #[case("a:a,b:d,e", "empty range")]
+    // end before start
+    #[case("b:a", "end of range is before start")]
+    // no start
+    #[case(":a", "at least one character, got none")]
+    // no end
+    #[case("a:", "at least one character, got none")]
+    // too many elements
+    #[case("a:b:e", "exactly 2 elements, got 3")]
+    fn selected_columns_from_invalid_ranges(#[case] raw: &str, #[case] message: &str) {
```
Comment (PrettyWood) [inline src/types/excelsheet.rs:770]: 👍

File: python/fastexcel/__init__.py [hunk 5]
```diff
@@ -129,8 +129,10 @@ def load_sheet_by_name(
...
+                            will select columns `A,C,D,F`.
```
Comment (PrettyWood) [inline python/fastexcel/__init__.py:135]: I suggest to right it with bullets
```
can either be:
- `None` to select all columns
- a list of strings, the column names
- a list of ints, the column indices (starting at 0)
- a string, a comma separated list of Excel column letters and column ranges (e.g. “A:E” or “A,C,E:F”)

**Review by PrettyWood — CHANGES_REQUESTED** [review]
Lol I started the PR this morning but didn't have time to finish. You're too fast boiiiiii

File: python/fastexcel/__init__.py (hunk 1 above)
Comment (lukapeschke) [reply python/fastexcel/__init__.py:178]: So `A,C,D,E,F` ?

File: src/types/excelsheet.rs (hunk 3 above)
Comment (lukapeschke) [reply src/types/excelsheet.rs:252]: woops forgot to remove the comment :sweat_smile:

File: python/fastexcel/__init__.py (hunk 1 above)
Comment (PrettyWood) [reply python/fastexcel/__init__.py:178]: exactly

Comment (lukapeschke) [reply python/fastexcel/__init__.py]: 248acdd

File: src/types/excelsheet.rs (hunk 3 above)
Comment (lukapeschke) [reply src/types/excelsheet.rs]: bb18d94

File: python/fastexcel/__init__.py (hunk 5 above)
Comment (lukapeschke) [reply python/fastexcel/__init__.py]: 47b815d

**Review by PrettyWood — APPROVED** [review]
Great work

**Pull request merged**